addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,272 @@
+        });
+
+        _.each(this.$('div.o_survey_form_simple_radio'), function (radio_group) {
+            self._updateSimpleRadioForDisplay($(radio_group));
+        });
+
+        _.each(this.$('div.o_survey_form_simple_select'), function (select_group) {
+            self._updateSimpleSelectForDisplay($(select_group));
+        });
+
+        return this._super.apply(this, arguments);
+    },
+
+    // -------------------------------------------------------------------------
+    // Private
+    // -------------------------------------------------------------------------
+
+    // EVENTS
+    // -------------------------------------------------------------------------
+    _onChangeRadio: function (event) {
+        var $radio_group = $(event.currentTarget).parents('.o_survey_form_simple_radio');
+        return this._updateSimpleRadioForDisplay($radio_group);
+    },
+
+    _onChangeSelect: function (event) {
+        var $select_group = $(event.currentTarget).parents('.o_survey_form_simple_select');
+        return this._updateSimpleSelectForDisplay($select_group);
+    },
+
+    _onChangeMultipleCheckbox: function (event) {
+        var $elem = $(event.currentTarget);
+        var $comment_input = $elem.parents('.o_survey_form_multiple').find('input[type="text"]');
+        if ($elem.prop('checked')) {
+            $comment_input.focus();
+        } else {
+            $comment_input.val('');
+        }
+    },
+
+    _onSubmit: function (event) {
+        var self = this;
+        event.preventDefault();
+        var $elem = $(event.currentTarget);
+        var submitValue = $elem.val();
+        var $form = $elem.parents('form.o_survey_form');
+        var formData = $form.serializeArray();
+        var params = {
+            button_submit: submitValue,
+        };
+        _.each(formData, function(data, idx){
+             params[data.name] = data.value;
+        })
+
+        this._resetErrors();
+
+        _.each($form.find('div.o_survey_form_date'), function (date_field) {
+            self._updateDateForSubmit($(date_field), params);
+        });
+
+        return rpc.query({
+            route: '/survey/validate/' + self.surveyToken + '/' + self.answerToken,
+            params: params,
+        }).then(function (result) {
+            if (result === true) {
+                rpc.query({
+                    route: '/survey/submit/' + self.surveyToken + '/' + self.answerToken,
+                    params: params
+                }).then(function (result) {
+                    window.location = result;
+                });
+            }
+            else if (result && result.fields && result.error === 'validation') {
+                _.each(_.keys(result.fields), function (key) {
+                    $("#" + key + '>.js_errzone').append('<p>' + result.fields[key] + '</p>').show();
+                    if (_.keys(result.fields)[_.keys(result.fields).length - 1] === key) {
+                         $('html, body').animate({
```

Comment:
Maybe extract to a method `_scrollToError` that takes the target as argument as this is used twice in this method.